apps/omg_eth/lib/omg_eth/root_chain/decode.ex

```diff
@@ -0,0 +1,527 @@
+    |> Enum.into(%{}, fn {key, _type, _indexed, value} -> {key, value} end)
+    |> HackNaming.in_flight_exit_blocked()
+    |> common_parse_event(log)
+  end
+
+  def in_flight_exit_finalized(log) do
+    function_spec = [in_flight_exit_input_withdrawn(), in_flight_exit_output_withdrawn()]
+
+    {_function_spec, data} =
+      ABI.Event.find_and_decode(
+        function_spec,
+        OMG.Eth.Encoding.from_hex(Enum.at(log["topics"], 0)),
+        OMG.Eth.Encoding.from_hex(Enum.at(log["topics"], 1)),
+        nil,
+        nil,
+        OMG.Eth.Encoding.from_hex(log["data"])
+      )
+
+    data
+    |> Enum.into(%{}, fn {key, _type, _indexed, value} -> {key, value} end)
+    |> HackNaming.in_flight_exit_finalized()
+    |> common_parse_event(log)
+  end
+
+  def in_flight_exit_started(log) do
+    function_spec = in_flight_exit_started()
+
+    {_function_spec, data} =
+      ABI.Event.find_and_decode(
+        [function_spec],
+        Encoding.from_hex(Enum.at(log["topics"], 0)),
+        Encoding.from_hex(Enum.at(log["topics"], 1)),
+        Encoding.from_hex(Enum.at(log["topics"], 2)),
+        nil,
+        Encoding.from_hex(log["data"])
+      )
+
+    data
+    |> Enum.into(%{}, fn {key, _type, _indexed, value} -> {key, value} end)
+    |> HackNaming.in_flight_exit_started()
+    |> common_parse_event(log)
+  end
+
+  def start_standard_exit(log) do
+    function_spec = start_standard_exit()
+
+    data =
+      function_spec
+      |> ABI.decode(log)
+      |> hd()
+      |> Tuple.to_list()
+
+    function_spec.input_names
+    |> Enum.zip(data)
+    |> Enum.into(%{})
+    |> HackNaming.start_standard_exit()
+  end
+
+  def start_in_flight_exit(log) do
+    function_spec = start_in_flight_exit()
+
+    data =
+      function_spec
+      |> ABI.decode(log)
+      |> hd()
+      |> Tuple.to_list()
+
+    function_spec.input_names
+    |> Enum.zip(data)
+    |> Enum.into(%{})
+    |> HackNaming.start_in_flight_exit()
+  end
+
+  #
+  # selector definitions
+  #
+  defp deposit_created() do
+    %ABI.FunctionSelector{
+      function: "DepositCreated",
+      input_names: ["depositor", "blknum", "token", "amount"],
+      inputs_indexed: [true, true, true, false],
+      method_id: <<24, 86, 145, 34>>,
+      returns: [],
+      type: :event,
+      types: [:address, {:uint, 256}, :address, {:uint, 256}]
+    }
+  end
+
+  defp in_flight_exit_input_piggybacked() do
+    %ABI.FunctionSelector{
+      function: "InFlightExitInputPiggybacked",
+      input_names: ["exitTarget", "txHash", "inputIndex"],
+      inputs_indexed: [true, true, false],
+      method_id: <<169, 60, 14, 155>>,
+      returns: [],
+      type: :event,
+      types: [:address, {:bytes, 32}, {:uint, 16}]
+    }
+  end
+
+  defp in_flight_exit_output_piggybacked() do
+    %ABI.FunctionSelector{
+      function: "InFlightExitOutputPiggybacked",
+      input_names: ["exitTarget", "txHash", "outputIndex"],
+      inputs_indexed: [true, true, false],
+      method_id: "n͎y",
```

Comment:
yeah this is just how binaries are represented sadly 
```
<<method_id::binary-size(4), _::binary>> = :keccakf1600.hash(:sha3_256, "InFlightExitChallengeResponded(address,bytes32,uint256)")
```

```
IO.inspect method_id, binaries: :as_binaries
<<110, 205, 142, 121>>
"n͎y"
```

Reply:
well yeah, I am aware of this, the thing to consider here is to write it down as `<<110, ...>>` in this file, since the unicode string there just looks super weird.